packages/bff/src/model/api/converters/catalogClientApiConverter.ts

```diff
@@ -0,0 +1,214 @@
+  document: catalogApi.EServiceDoc
+): bffApi.EServiceDoc {
+  return {
+    id: document.id,
+    name: document.name,
+    contentType: document.contentType,
+    prettyName: document.prettyName,
+  };
+}
+
+export function toBffCatalogApiEserviceRiskAnalysis(
+  riskAnalysis: catalogApi.EServiceRiskAnalysis
+): bffApi.EServiceRiskAnalysis {
+  const answers: { [key: string]: string[] } =
+    riskAnalysis.riskAnalysisForm.singleAnswers
+      .concat(
+        riskAnalysis.riskAnalysisForm.multiAnswers.flatMap((multiAnswer) =>
+          multiAnswer.values.map((answerValue) => ({
+            id: multiAnswer.id,
+            value: answerValue,
+            key: multiAnswer.key,
+          }))
+        )
+      )
+      .reduce((answers: { [key: string]: string[] }, answer) => {
+        const key = `${answer.key}`;
+        if (answers[key] && answer.value) {
+          answers[key] = [...answers[key], answer.value];
+        } else {
+          answers[key] = [];
+        }
+
+        return answers;
+      }, {});
+
+  const riskAnalysisForm: bffApi.RiskAnalysisForm = {
+    riskAnalysisId: riskAnalysis.id,
+    version: riskAnalysis.riskAnalysisForm.version,
+    answers,
+  };
+
+  return {
+    id: riskAnalysis.id,
+    name: riskAnalysis.name,
+    createdAt: riskAnalysis.createdAt,
+    riskAnalysisForm,
+  };
+}
+
+export function toBffCatalogApiProducerDescriptorEService(
+  eservice: catalogApi.EService,
+  producer: tenantApi.Tenant
+): bffApi.ProducerDescriptorEService {
+  const producerMail = producer.mails.find(
+    (m) => m.kind === tenantMailKind.ContactEmail
+  );
+
+  const notDraftDecriptors: bffApi.CompactDescriptor[] =
+    eservice.descriptors.filter((d) => d.state !== descriptorApiState.DRAFT);
+
+  const draftDescriptor: bffApi.CompactDescriptor | undefined =
+    eservice.descriptors.find((d) => d.state === descriptorApiState.DRAFT);
+
+  return {
+    id: eservice.id,
+    name: eservice.name,
+    description: eservice.description,
+    technology: eservice.technology,
+    mode: eservice.mode,
+    mail: producerMail && {
+      address: producerMail.address,
+      description: producerMail.description,
+    },
+    draftDescriptor,
+    riskAnalysis: eservice.riskAnalysis.map(
+      toBffCatalogApiEserviceRiskAnalysis
+    ),
+    descriptors: notDraftDecriptors,
+  };
+}
+
+export function toEserviceAttribute(
+  attributes: catalogApi.Attribute[]
+): EServiceAttribute[] {
+  return attributes.map((attribute) => ({
+    ...attribute,
+    id: unsafeBrandId(attribute.id),
+  }));
+}
+
+export function toDescriptorWithOnlyAttributes(
+  descriptor: catalogApi.EServiceDescriptor
+): DescriptorWithOnlyAttributes {
+  return {
+    ...descriptor,
+    attributes: {
+      certified: descriptor.attributes.certified.map(toEserviceAttribute),
+      declared: descriptor.attributes.declared.map(toEserviceAttribute),
+      verified: descriptor.attributes.verified.map(toEserviceAttribute),
+    },
+  };
+}
```

Comment:
We are using runtime resources to change something at type level, I know the performance impact could be negligible but I saw this in a lot of part of the monorepo and I'm wondering if it is worth discussing. 
What do you think?

Reply:
@Carminepo2 good point 👍🏻 
unfortunately branded type usages requires mapping like this, so we need to choose what we want encourage. Id as string are very error prone when they are function's parameter, branded type is a sort of guard to avoid bugs, meanwhile every time we convert models it's necessary this wrapping... 
I suggest to discuss a possible policy offline.